src/index.js

```diff
@@ -136,17 +207,107 @@ module.exports = {
           (sum, userResult) => sum + userResult.data.betaVector[i], 0
         ) / userCount);
       }
+      
+      // calculate globalMeanY
+      const siteCount=userResults.length;  
+      var totalY=0;
+      var globalYCount=0;
+       
+      for (let i=0; i < siteCount; i += 1) {
+         totalY += userResults[i].data.localMeanY*userResults[i].data.localCount;
+         globalYCount += userResults[i].data.localCount;
+      } 
 
+      const globalMeanY = totalY/globalYCount;
+   
       /* eslint-disable no-console */
       console.log('Average beta vector:', averageBetaVector);
+      console.log('globalMeanY is :', globalMeanY);
       /* eslint-enable no-console */
+ 
+      // Extract X and y variable name list from userData
+      const y_label=opts.userResults[0].userData.__FEATURES__;
+      const X_label=Object.keys(opts.userResults[0].userData.files[0].tags);
+      console.log('X_label is', X_label);
+      console.log('y_label is',y_label);
 
       return {
         averageBetaVector,
+        globalMeanY,
+        X_label,
+        y_label,
+//        complete: true,
+      };
+    },
+    verbose: true,
+  }, {
+    // get sstNode from each local node and calculate the statistics
+    type: 'function',
+    fn(opts) {
+
+    const userResults = opts.userResults;
```

Comment:
Once again, destructuring can clean this up:

```js
const {
  previousData: { X_label, y_label },
  userResults,
  userResults: [{
    data: { averageBetaVector },
  }],
} = opts;
```

Also worth noting: our style guide prohibits variable names with an underscore. You should rewrite the variables and properties using camelCase.